ludwig/trainers/trainer.py

```diff
@@ -1200,6 +1202,9 @@ def _train_loop(
                     if self.is_coordinator():
                         progress_tracker.save(os.path.join(save_path, TRAINING_PROGRESS_TRACKER_FILE_NAME))
 
+                # Callback that the checkpoint was reached, regardless of whether the model was evaluated or saved.
+                self.callback(lambda c: c.on_checkpoint(self, progress_tracker))
```

Comment:
This should be in the scope of the `if not self.skip_save_progress`, otherwise it seems misleading (since there was no checkpoint written out).

Reply:
+1.